src/github.com/getlantern/enproxy/conn_impl.go

```diff
@@ -38,12 +37,18 @@ func Dial(addr string, config *Config) (*Conn, error) {
 		return nil, fmt.Errorf("Unable to dial proxy: %s", err)
 	}
 
+	go c.processWrites()
+	go c.processReads()
 	go c.processRequests(proxyConn)
```

Comment:
All goroutines are now spawned after the initial dial to the proxy.  That way, if the initial dial failed and we return an error, we don't have to clean up after them.